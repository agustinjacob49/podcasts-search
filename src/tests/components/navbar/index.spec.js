/* eslint-disable testing-library/no-unnecessary-act */
/* eslint-disable testing-library/prefer-screen-queries */
import { fireEvent, render, waitFor } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Navbar from "../../../components/navbar";
import * as router from 'react-router';
import LoaderContext from "../../../utils/context/loaderContext";


const navigate = jest.fn();

beforeEach(() => {
  jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
})

jest.setTimeout(900000);
describe("Navbar", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  afterAll( () => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });

  test("Snapshot testing - Should render podcaster", async () => {

    const { container, getByText, queryAllByTestId } = render(<Router><Navbar /></Router>);

    expect(container).toMatchSnapshot();
    expect(getByText("Podcaster")).toBeDefined();
    expect(queryAllByTestId("loader")).toStrictEqual([]);
  });

  test("Snapshot testing - Should render loader", async () => {

    const { container, getAllByTestId } = render(<Router> <LoaderContext.Provider value={true}><Navbar /> </LoaderContext.Provider></Router>);

    expect(container).toMatchSnapshot();
    expect(getAllByTestId("loader")).toBeDefined();
  });
});
