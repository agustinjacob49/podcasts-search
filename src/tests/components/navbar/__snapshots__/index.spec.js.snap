// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot testing - Should render loader 1`] = `
<div>
   
  <header>
    <nav
      class="navbar"
    >
      <a
        class="navbar__title"
        data-testid="podcaster"
        href="/"
      >
         Podcaster
      </a>
      <div
        class="navbar__loader"
        data-testid="loader"
      >
         
        <div
          class="loader"
        />
         
      </div>
    </nav>
  </header>
   
</div>
`;

exports[`Navbar Snapshot testing - Should render podcaster 1`] = `
<div>
  <header>
    <nav
      class="navbar"
    >
      <a
        class="navbar__title"
        data-testid="podcaster"
        href="/"
      >
         Podcaster
      </a>
    </nav>
  </header>
</div>
`;
